java/arcs/core/data/SchemaFields.kt

```diff
@@ -41,14 +43,29 @@ sealed class FieldType(
 
   data class ListOf(val primitiveType: FieldType) : FieldType(Tag.List)
 
+  data class NullableOf(val innerType: FieldType) : FieldType(Tag.Nullable) {
+    init {
+      if (!BuildFlags.NULLABLE_VALUE_SUPPORT) {
+        throw BuildFlagDisabledError("NULLABLE_VALUE_SUPPORT")
+      }
+    }
+  }
+
   data class InlineEntity(val schemaHash: String) : FieldType(Tag.InlineEntity)
 
   enum class Tag {
     Primitive,
     EntityRef,
     Tuple,
     List,
-    InlineEntity
+    InlineEntity,
+    Nullable
+  }
+
+  fun nullable(): FieldType = if (!BuildFlags.NULLABLE_VALUE_SUPPORT) {
```

Comment:
Could you add KDoc here explaining that it only returns a nullable if the field is enabled, and otherwise returns the value as-is. And also add a TODO linked to the tracking bug for the `NULLABLE_VALUE_SUPPORT` flag, as a reminder to fix this up after it has launched.

Reply:
Added, and filed a bug (blocked on Nullable Support) to track the flag clean up work (b/181084704)